### ISSUE TITLE
how about the current performance

### ISSUE THREAD

**Issue by gaopeng-eugene** [amdegroot/ssd.pytorch#4]

**Comment by christopher5106** [comment]
mAP score ?

**Comment by amdegroot** [comment]
Hey sorry just getting back to these, was on vacation this past weekend away from my computer.  We should have an official mAP score as well as timing metrics by Thursday of this week, hopefully sooner... We've just recently been caught up on a lot of other work, but stay tuned.

**Comment by oarriaga** [comment]
Hello @amdegroot do you have any info regarding the current mAP score?

**Comment by amdegroot** [comment]
Hey @oarriaga, 
Thanks for checking out the repo.  Unfortunately, I've been distracted by a lot of school work recently, but I'm still in the process of getting an official 'training from scratch' mAP score and should have it within the next week.  Currently I can confirm that the model with pre-trained weights reproduces the 77.26% mAP of the original caffe version, and training from scratch is close.  We just need to finish data augmentation, and combining VOC2007 and VOC2012 for training (currently the repo is set up to only train on one or the other).   Any and all contributions are welcome, so feel free to submit a PR, and the repo is updated fairly frequently so make sure to stay tuned.

**Comment by gaopeng-eugene** [comment]
How about the FPS of current model? This is another very important factor.

**Comment by amdegroot** [comment]
@gaopeng-eugene You're right, I will post this as soon as I can.  I am currently maxing out my GTX 1060 for training right now, but to give you an idea, it was taking ~0.022s per image during evaluation (on GTX 1060), so it should be well be above 50 fps for just detection.  It should also be easy to benchmark yourself.  I would love to hear results from someone with a TitanX.

**Comment by amdegroot** [comment]
After benchmarking the current FPS on a GTX 1060, it looks like we're at 43.47 FPS for detection on a single image.

**Comment by gaopeng-eugene** [comment]
Nice to hear the results. Thank you for your contribution.

**Comment by gaopeng-eugene** [comment]
The mAP is 50.8 when training from scratch? Right ?

**Comment by amdegroot** [comment]
For now, without the data augmentation, correct.

**Comment by amdegroot** [comment]
Actually, I'm training right now and with a different learning rate, at 15k iterations I'm already at 54.7% mAP (still w/o data augmentation), so hopefully once this finishes training it'll be a lot better than before.

**Comment by xwjabc** [comment]
What is the mAP for pretrained models? I am trying to migrate mAP calculation module from longcw's repo for Faster RCNN. Is there any possibility to offer the mAP calculation part?

**Comment by amdegroot** [comment]
Just posted the link to Pytorch trained from scratch weights that achieve mAP of 77.43%.  Also added the mAP eval code used to calculate that result.

**Comment by acrosson** [comment]
Just downloaded and trained for 10k iterations. The evaluation script is taking ~0.30 seconds / image. Are you still seeing 45 FPS?

**Comment by isalirezag** [comment]
when i run the test net, it does not give me the final mAP, how can i compute that 
:(

**Comment by liuchang8am** [comment]
Same with acrosson here, inference time is about 300ms per image on GTX 1070, which is far slower than the 45FPS in the description.

**Comment by SystemErrorWang** [comment]
I met the same problem. When using original vgg backbone the speed is ok. I then changed the backbone to resnet101, the training speed did not slow down much but testing for each image took 2 seconds now.

**Comment by d-li14** [comment]
Same issue about speed much lower than 45fps. And @SystemErrorWang, how long did you take when trying the original vgg16-based model, up to 45fps? My evaluation process will take about 0.1s even with the provided pretrained vgg model.

**Comment by CasonTsai** [comment]
@acrosson i met the same situation.Have you solved this problem?
